### PR TITLE
Feature/add favorite

### PR DESCRIPTION
**Pull request by YuZhangYuanYYZ**

### CODE REVIEW

File: src/store/reducers/agentsReducer.js [hunk 1]
```diff
@@ -80,6 +85,27 @@ export function agentsReducer(state = initialState, action) {
...
+      return {
+        ...state,
+        toggleFavoriteAgentInprogress: false,
+        items: state.items.map((item) => {
+          if (item.id !== action.payload.agentId) {
+            item.isFavorite = !item.isFavorite;
+          }
+          return item;
+        }),
+      };
```
Comment (cxss-engine-user) [inline src/store/reducers/agentsReducer.js:103]:
```suggestion
     case TOGGLE_FAVORITE_SUCCESS:
      return {
        ...state,
        toggleFavoriteAgentInprogress: false,
        items: state.items.map((item) => {
          if (item.id == action.payload.agentId) {
            return  action.payload
          }
          return item;
        }),
      };
```

File: src/store/reducers/agentsReducer.js [hunk 2]
```diff
@@ -80,6 +85,27 @@ export function agentsReducer(state = initialState, action) {
...
+    case TOGGLE_FAVORITE_FAIL:
```
Comment (lanmengyujing) [inline src/store/reducers/agentsReducer.js:107]:
```
case TOGGLE_FAVORITE_SUCCESS:
       return {
         ...state,
         toggleFavoriteAgentInprogress: false,
         items: state.items.map((item) => {
           if (item.id === action.payload.agentId) {
             return action.payload;
           }
           return item;
         }),
       };
```

File: src/components/MainContent/AgentList/AgentItem/AgentItem.js [hunk 3]
```diff
@@ -41,6 +42,8 @@ export function AgentItem({
           resources={resources}
           status={status}
           setIsComponentVisible={setIsComponentVisible}
+          isFavorite={agent.isFavorite}
```
Comment (YuZhangYuanYYZ) [inline src/components/MainContent/AgentList/AgentItem/AgentItem.js:45]: delete

File: src/components/MainContent/AgentList/AgentItem/AgentResources/AgentResources.js [hunk 4]
```diff
@@ -10,6 +11,8 @@ export function AgentResources({
   status,
   resources,
   setIsComponentVisible,
+  isFavorite,
```
Comment (YuZhangYuanYYZ) [inline src/components/MainContent/AgentList/AgentItem/AgentResources/AgentResources.js]: delete

File: src/components/MainContent/AgentList/AgentList.js [hunk 5]
```diff
@@ -42,6 +42,8 @@ export function AgentList({ agents }) {
                   resources={agent.resources}
                   setIsComponentVisible={setIsComponentVisible}
                   agentIndex={index}
+                  isFavorite={agent.isFavorite}
```
Comment (YuZhangYuanYYZ) [inline src/components/MainContent/AgentList/AgentList.js:45]: delete

**Pull request closed without merging**